content/tutorials/WebRTC-file-transfers-with-Ably.textile

```diff
@@ -0,0 +1,509 @@
+
+**What happens in the onReadAsDataURL method?**
+
+```[javascript]
+    function onReadAsDataURL(event, text, file) {
+        var chunkLength = 1024;
+        var data = {}; // data object to transmit over data channel
+        data.name = file.name
+        data.size = file.size
+        if (event) {
+            text = event.target.result; // on first invocationgettrue;we 
+            data.stringSize = text.length
+            messageList[currentChat][file.name] = {
+                completed: true,
+                sender: 'me',
+                downloadLink: text,
+                chunks: []
+            }
+            render()
+        }
+        if (text.length > chunkLength) {
+            data.message = text.slice(0, chunkLength); // getting chunk using predefined chunk length
+        } else {
+            data.message = text;
+            data.last = true;
+        }
+        data.sender = clientId
+        connections[currentChat].send(JSON.stringify(data));
+        var remainingDataURL = text.slice(data.message.length);
+        if (remainingDataURL.length) {
+            onReadAsDataURL(null, remainingDataURL, file); // continue transmitting
+        }
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/a5a88efb0e6accc721d7f27a211874b528b303bf
+
+The method above accepts three parameters which are:
+
+
+# @event@: the instance of the event when the @fileReader@ was initially loaded. This would be passed the first time @fileReader@ initiates the method.
+# @text@: this refers to the remaining part of the data that was chunked. This would be passed so far there is still data pertaining to the file being sent.
+# @file@: this refers to an instance of the file being sent.
+
+We first define the limit for chunks which we will be sending in batches. Here, we want to send only 1024 bytes at a time.
+
+Next, we declare an object called data, which will contain metadata of the file we want to send.
+
+The @if(event)@ block caters for instances when it is initiated by the @fileReader@  itself, and not the remaining part of the file. In this block, we set the @text@ variable to the result of the @fileReader@, which is a DataUrl string. We then push a new object key, with the name of the file we want to send to our file conversation with the other peer.
+
+Next, we check if the length of the DataUrl is greater than the defined chunk size. If true, we slice the text, else, we send the text, with an extra property telling the other peer this is the last part of the data. This is useful to know when all parts of the file have been received.
+
+h2.
+  default: Step 7 – Receiving files
+
+Receiving the files consist of storing the chunks in an array, then joining the chunks together when all parts have been received. Let’s take a look at the @recieveMessage@ function below:
```

Comment:
recieve typo - please also correct these in code samples uploaded in the tutorials repo.